dxr/plugins/omniglot.py

```diff
@@ -192,12 +192,16 @@ def synth_web_url(self, repo):
         if repo.startswith("git@github.com:"):
             self._is_github = True
             return "https://github.com/" + repo[len("git@github.com:"):]
-        elif repo.startswith("git://github.com/"):
+        elif repo.startswith(("git://github.com/", "https://github.com/")):
             self._is_github = True
             if repo.endswith(".git"):
                 repo = repo[:-len(".git")]
-            return "https" + repo[len("git"):]
-        raise Exception("I don't know what's going on")
+            if repo.startswith("git:"):
+                repo = "https" + repo[len("git"):]
+            return repo
+        else:
```

Comment:
The fix looks good to me. And I see where you were going with the `else` here; it's very noble. :-) But the rest of omniglot seems to assume something useful will be in `self.upstream`, and, if we just make up a URL, it won't be. So let's keep raising an exception in that case, with a message apologizing that we only support GitHub-derived repos right now. Ultimately, we should provide a config value so you can specify a URL manually, but first I want to refactor the config system.